src/NServiceBus.RavenDB.Tests/Outbox/When_adding_outbox_messages.cs

```diff
@@ -99,5 +100,81 @@ public void Should_update_dispatched_flag()
+        [TestCase("Outbox/TestEndpoint/")]
+        public void Should_get_messages_with_old_and_new_recordId_format(string outboxRecordIdPrefix)
+        {
+            var sessionFactory = new RavenSessionFactory(store);
+            var persister = new OutboxPersister(sessionFactory) { DocumentStore = store, EndpointName = "TestEndpoint" };
+
+            var messageId = Guid.NewGuid().ToString();
+
+            //manually store an OutboxRecord to control the OutboxRecordId format
+            sessionFactory.Session.Store(new OutboxRecord
+            {
+                MessageId = messageId,
+                Dispatched = false,
+                TransportOperations = new List<OutboxRecord.OutboxOperation>
+                {
+                    new OutboxRecord.OutboxOperation
+                    {
+                        Message = new byte[1024 * 5],
+                        Headers = new Dictionary<string, string>(),
+                        MessageId = messageId,
+                        Options = new Dictionary<string, string>()
+                    }
+                }
+            }, outboxRecordIdPrefix + messageId);
+
+            sessionFactory.SaveChanges();
+            sessionFactory.ReleaseSession();
+
+            OutboxMessage result;
+            persister.TryGet(messageId, out result);
+
+            Assert.NotNull(result);
+            Assert.AreEqual(messageId, result.MessageId);
+        }
+
+        [TestCase("Outbox/")]
+        [TestCase("Outbox/TestEndpoint/")]
+        public void Should_set_messages_as_dispatched_with_old_and_new_recordId_format(string outboxRecordIdPrefix)
+        {
+            var sessionFactory = new RavenSessionFactory(store);
+            var persister = new OutboxPersister(sessionFactory) { DocumentStore = store, EndpointName = "TestEndpoint" };
+
+            var messageId = Guid.NewGuid().ToString();
+
+            //manually store an OutboxRecord to control the OutboxRecordId format
+            sessionFactory.Session.Store(new OutboxRecord
+            {
+                MessageId = messageId,
+                Dispatched = false,
+                TransportOperations = new List<OutboxRecord.OutboxOperation>
+                {
+                    new OutboxRecord.OutboxOperation
+                    {
+                        Message = new byte[1024 * 5],
+                        Headers = new Dictionary<string, string>(),
+                        MessageId = messageId,
+                        Options = new Dictionary<string, string>()
+                    }
+                }
+            }, outboxRecordIdPrefix + messageId);
+
+            sessionFactory.SaveChanges();
+            sessionFactory.ReleaseSession();
+
+            persister.SetAsDispatched(messageId);
+            sessionFactory.ReleaseSession();
+
+            WaitForIndexing(store);
+
+            var result = sessionFactory.Session.Query<OutboxRecord>().Where(o => o.MessageId == messageId).SingleOrDefault();
```

Comment:
To be honest, it's because I copied most of that from the pre-existing `Should_update_dispatched_flag()` test, and it uses the index to query also.

The original test used the query because it called `Store()` so it didn't have the `OutboxRecordId` to be able to `Load()` it. 

However, in my test, I do know what the `OutboxRecordId` is, so I can just load it. I'll go ahead and change that.